cli/install.go

```diff
@@ -205,8 +205,62 @@ type githubTaggedReleaseResponse struct {
 }
 
 type gitHubAsset struct {
-	Url  string // download url for asset
-	Name string // contains platform information about binary
+	Url                  string // url for asset json object
+	Name                 string // contains platform information about binary
+	Browser_download_url string // download url
+}
+
+func GetZlsDownloadUrl(version string, archDouble string) (string, error) {
+	if version == "master" {
+		var exeName = "zls"
+		if strings.Contains(archDouble, "windows") {
+			exeName = "zls.exe"
+		}
+		return fmt.Sprintf("https://zig.pm/zls/downloads/%v/bin/%v", archDouble, exeName), nil
+
+	} else {
+		var releaseUrl string
+		if version == "master" {
+			releaseUrl = "https://api.github.com/repos/zigtools/zls/releases/latest"
+		} else {
+			// build url for tagged version
+			releaseUrl = fmt.Sprintf("https://api.github.com/repos/zigtools/zls/releases/tags/%v", version)
+		}
+
+		// get release information
+		resp, err := http.Get(releaseUrl)
+		if err != nil {
+			return "", err
+		}
+		defer resp.Body.Close()
+
+		var releaseBuffer bytes.Buffer
+		_, err = releaseBuffer.ReadFrom(resp.Body)
+		if err != nil {
+			return "", err
+		}
+
+		// getting list of assets
+		var taggedReleaseResponse githubTaggedReleaseResponse
+		if err := json.Unmarshal(releaseBuffer.Bytes(), &taggedReleaseResponse); err != nil {
+			return "", err
+		}
+
+		// getting platform information
+		downloadUrl := ""
```

Comment:
This variable could be `var downloadURL string`, but you could keep it like this.